torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2989,34 +3001,45 @@ such as `dist.all_reduce(tensor, async_op=True)`.
                        true when tensors have arrived on respective nodes, but not yet necessarily synched on
                        respective GPUs (similarly to GPU work).
            )")
-      .def(
-          "_get_op_type",
-          [](::c10d::Work& work) -> int {
-            return static_cast<int>(work.retrieveOpType());
-          })
-      .def(
-          "_get_duration",
-          &::c10d::Work::getDuration,
-          py::call_guard<py::gil_scoped_release>(),
-          R"(
+          .def(
+              "_get_op_type",
+              [](::c10d::Work& work) -> int {
+                return static_cast<int>(work.retrieveOpType());
+              })
+          .def(
+              "_get_duration",
+              &::c10d::Work::getDuration,
+              py::call_guard<py::gil_scoped_release>(),
+              R"(
               Returns:
                   Duration of the corresponding collective communication.
 
               .. warning ::
                   This API only works for NCCL backend for now and must set
                   TORCH_NCCL_ENABLE_TIMING environment variable.
             )")
-      .def(
-          "boxed",
-          [](c10::intrusive_ptr<::c10d::Work> self) {
-            return torch::jit::toPyObject(c10::IValue(std::move(self)));
-          })
-      .def_static("unbox", [](py::object obj) {
-        auto typePtr =
-            torch::getCustomClass("__torch__.torch.classes.c10d.Work");
-        auto ivalue = torch::jit::toIValue(std::move(obj), typePtr);
-        return ivalue.toCustomClass<::c10d::Work>();
-      });
+          .def(
+              "boxed",
+              [](c10::intrusive_ptr<::c10d::Work> self) {
+                return torch::jit::toPyObject(c10::IValue(std::move(self)));
+              })
+          .def_static("unbox", [](py::object obj) {
+            auto typePtr =
+                torch::getCustomClass("__torch__.torch.classes.c10d.Work");
+            auto ivalue = torch::jit::toIValue(std::move(obj), typePtr);
+            return ivalue.toCustomClass<::c10d::Work>();
+          });
+
+#ifdef USE_C10D_NCCL
+  py::class_<
+      ::c10d::ProcessGroupNCCL::WorkNCCL,
+      std::shared_ptr<::c10d::ProcessGroupNCCL::WorkNCCL>>(
+      processGroupNCCL,
+      "WorkNCCL");
+//       R"(
+// A `WorkNCCL` object represents the handle to a `Work` object issued on GPU.)")
+    //   .def("timeout", &::c10d::ProcessGroupNCCL::WorkNCCL::getOpTimeout);
+#endif
```

Comment:
I need to expose WorkNCCL via pybind, but as long as I have this piece of code, it will just fail init of c10d...

Reply:
why do you need to expose Work via pybind?

it makes me nervous because we already expose Work but maybe its not via pybind- maybe its via torchbind (the thing from torchscript interpreter)?

I ran into some gnarly issues with binding incompatibility when working on functional collectives.  @howard was helpful.  In any case, i'm hoping you can just avoid having to bind WorkNCCL?

Reply:
Well, I implicitly check the timeout change is expected, but we need a way to check the timeout of a Work? Maybe I can create an API inside NCCLPG?

Reply:
> @howard was helpful

I think you mean someone else 😉

Reply:
cc: @H-Huang